### PR TITLE
doc: net: Add Thread protocol description

### PR DESCRIPTION
**Pull request by rlubos**
Nothing fancy, mostly gathering some useful links in one place.

I did not add a reference to this page yet, as I wanted to wait for #13616 first (hence DNM). Once #13616 is merged, I'll rebase update the PR.

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

File: doc/reference/networking/thread.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+and all communications within the mesh are encrypted. More information about
+Thread protocol can be found at
+`Thread Group website <https://www.threadgroup.org>`_.
+
+Zephyr integrates an open source Thread protocol implementation called OpenThread.
+See `OpenThread website <https://openthread.io/>`_ for more information about
+the OpenThread itself.
+
+Internet connectivity
+=====================
```
Comment (jukkar) [inline doc/reference/networking/thread.rst]: I just wonder should the chapter use `*******` here, at least the other docs the Overview + other chapters are in the same level.

File: doc/reference/networking/thread.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+A Thread Border Router is required to connect mesh network to the internet.
+An open source implementation of Thread Border Router is provided by OpenThread
+community. See
+`OpenThread Border Router guide <https://openthread.io/guides/border-router>`_
+for instructions on how to setup a Border Router.
+
+Sample usage
+============
```
Comment (jukkar) [inline doc/reference/networking/thread.rst]: ditto

**Review by jukkar — APPROVED** [review]
The content is ok so +1 from me.

File: doc/reference/networking/thread.rst (hunk 1 above)
Comment (rlubos) [reply doc/reference/networking/thread.rst]: Right, let's be consistent here.

File: doc/reference/networking/thread.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+technology over IEEE 802.15.4 protocol. IP connectivity allows to easily connect
+a Thread mesh network to the internet with a Thread Border Router.
+
+Thread specification provides a high level of network security. Mesh networks
+built with Thread are secure - only authenticated devices can join the network
+and all communications within the mesh are encrypted. More information about
+Thread protocol can be found at
+`Thread Group website <https://www.threadgroup.org>`_.
+
+Zephyr integrates an open source Thread protocol implementation called OpenThread.
```
Comment (dbkinder) [inline doc/reference/networking/thread.rst]: How about:
```
Zephyr integrates an open source Thread protocol implementation called OpenThread,
documented on the `OpenThread website <https://openthread.io/>`_.
```

File: doc/reference/networking/thread.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+See `OpenThread website <https://openthread.io/>`_ for more information about
+the OpenThread itself.
+
+Internet connectivity
+*********************
+
+A Thread Border Router is required to connect mesh network to the internet.
+An open source implementation of Thread Border Router is provided by OpenThread
```
Comment (dbkinder) [inline doc/reference/networking/thread.rst]:
```suggestion
An open source implementation of Thread Border Router is provided by the OpenThread
```

File: doc/reference/networking/thread.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+community. See
+`OpenThread Border Router guide <https://openthread.io/guides/border-router>`_
+for instructions on how to setup a Border Router.
```
Comment (dbkinder) [inline doc/reference/networking/thread.rst]: "setup" is a noun, while "set up" is a verb phrase:
```suggestion
for instructions on how to set up a Border Router.
```

File: doc/reference/networking/thread.rst [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+Sample usage
+************
+
+You can try the OpenThread with Echo server and Echo client samples, which
```
Comment (dbkinder) [inline doc/reference/networking/thread.rst]:
```suggestion
You can try using OpenThread with the Zephyr Echo server and Echo client samples, which
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by rlubos** [comment]
Thank you, I've pushed the updated version.

**Review by dbkinder — APPROVED** [review]
Looks good, thanks!

**Pull request merged**